src/tbbbind/CMakeLists.txt

```diff
@@ -70,21 +70,24 @@ function(tbbbind_build TBBBIND_NAME REQUIRED_HWLOC_TARGET)
             ${TBB_LIB_LINK_FLAGS}
             ${TBB_COMMON_LINK_FLAGS}
             ${TBB_IPO_LINK_FLAGS}
-            ${HWLOC_LIBRARY_DIRS} # pkg-config defined
         )
     else()
         target_link_libraries(${TBBBIND_NAME}
             PRIVATE
             ${TBB_LIB_LINK_FLAGS}
             ${TBB_COMMON_LINK_FLAGS}
             ${TBB_IPO_LINK_FLAGS}
-            ${HWLOC_LIBRARY_DIRS} # pkg-config defined
         )
     endif()
 
+    # Get the full path to HWLOC if it was found by pkg-config
+    if (${HWLOC_TARGET_LINK_LIBRARIES} STREQUAL ${HWLOC_LIBRARIES})
+        find_library(HWLOC_TARGET_LINK_LIBRARIES ${HWLOC_LIBRARIES} PATHS ${HWLOC_LIBRARY_DIRS})
```

Comment:
From #498 

Will you have to duplicate it here https://github.com/oneapi-src/oneTBB/blob/ad4c9bcc47382858c88a9ea58b7c3416072ea583/test/CMakeLists.txt#L266 ?

Maybe it makes sense to do such stuff somewhere near https://github.com/oneapi-src/oneTBB/blob/e96dbf4b48e689ab810b0428d0858a6abc2c83bd/cmake/hwloc_detection.cmake#L55 ?

I know you want to make the approach working for CMake 3.1, but just in case you can consider using pkg_check_modules with IMPORTED_TARGET (available since CMake 3.6). Or you can try creation of imported target on your own.

Reply:
Great idea, thanks!